### PR TITLE
Adding data flow request from cm to hip

### PR DESCRIPTION
**Pull request by Nexengineer**

### CODE REVIEW

File: src/main/java/in/projecteka/consentmanager/ConsentManagerConfiguration.java [hunk 1]
```diff
@@ -229,6 +231,23 @@ public PostDataFlowRequestApproval postDataFlowRequestApproval(AmqpTemplate amqp
...
+                                                                    DataFlowNotifier dataFlowNotifier,
+                                                                    DataFlowRequestRepository dataFlowRequestRepository,
+                                                                    WebClient.Builder builder,
+                                                                    ClientRegistryProperties clientRegistryProperties) {
+        return new DataFlowBroadcastListener(messageListenerContainerFactory,
+                destinationsConfig,
+                jackson2JsonMessageConverter,
+                dataFlowNotifier,
+                dataFlowRequestRepository,
+                new ClientRegistryClient(builder, clientRegistryProperties)
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/ConsentManagerConfiguration.java]:
```suggestion
                new ClientRegistryClient(builder, clientRegistryProperties));
```

File: src/main/java/in/projecteka/consentmanager/ConsentManagerConfiguration.java [hunk 2]
```diff
@@ -229,6 +231,23 @@ public PostDataFlowRequestApproval postDataFlowRequestApproval(AmqpTemplate amqp
...
+                );
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/ConsentManagerConfiguration.java]:
```suggestion
```

File: src/main/java/in/projecteka/consentmanager/clients/DataFlowNotifier.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import reactor.core.publisher.Mono;
+
+import static in.projecteka.consentmanager.clients.ClientError.unknownErrorOccurred;
+
+@AllArgsConstructor
+public class DataFlowNotifier {
+    private WebClient.Builder webClientBuilder;
+
+    public Mono<Void> notifyHip(DataFlowRequest dataFlowRequest,
+                                String hipUrl) {
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/clients/DataFlowNotifier.java]:
```suggestion
```

File: src/main/java/in/projecteka/consentmanager/clients/DataFlowNotifier.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import static in.projecteka.consentmanager.clients.ClientError.unknownErrorOccurred;
+
+@AllArgsConstructor
+public class DataFlowNotifier {
+    private WebClient.Builder webClientBuilder;
+
+    public Mono<Void> notifyHip(DataFlowRequest dataFlowRequest,
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/clients/DataFlowNotifier.java]:
```suggestion
    public Mono<Void> notifyHip(DataFlowRequest dataFlowRequest, String hipUrl) {
```

File: src/main/java/in/projecteka/consentmanager/ConsentManagerConfiguration.java (hunk 1 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/consentmanager/ConsentManagerConfiguration.java]: Done

File: src/main/java/in/projecteka/consentmanager/ConsentManagerConfiguration.java (hunk 2 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/consentmanager/ConsentManagerConfiguration.java]: Done

File: src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    private Mono<String> providerUrl(String providerId) {
+        return clientRegistryClient.providerWith(providerId)
+                .flatMap(provider -> provider.getIdentifiers()
+                        .stream()
+                        .filter(Identifier::isOfficial)
+                        .findFirst()
+                        .map(identifier -> Mono.just(identifier.getSystem()))
+                        .orElse(Mono.empty()));
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java]: Log the error if we can't find the URL of HIP, and find a way to put the request into dead-letter-queue.

File: src/main/java/in/projecteka/consentmanager/clients/DataFlowNotifier.java (hunk 3 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/consentmanager/clients/DataFlowNotifier.java]: Done

**Pull request merged**